core-image.html

```diff
@@ -195,49 +310,65 @@
       this.style.height = isNaN(this.height) ? this.height : this.height + 'px';
     },
 
-    update: function() {
-      this.style.backgroundSize = this.sizing;
-      this.style.backgroundPosition = this.sizing ? this.position : null;
-      this.style.backgroundRepeat = this.sizing ? 'no-repeat' : null;
-      if (this.preload) {
-        if (this.fade) {
-          if (!this._placeholderEl) {
-            this._placeholderEl = this.shadowRoot.querySelector('#placeholder');
-          }
-          this._placeholderEl.style.backgroundSize = this.sizing;
-          this._placeholderEl.style.backgroundPosition = this.sizing ? this.position : null;
-          this._placeholderEl.style.backgroundRepeat = this.sizing ? 'no-repeat' : null;
-          this._placeholderEl.classList.remove('fadein');
-          this._placeholderEl.style.backgroundImage = (this.load && this.placeholder) ? 'url(' + this.placeholder + ')': null;
-        } else {
-          this._setSrc(this.placeholder);
-        }
-        if (this.load && this.src) {
-          var img = new Image();
-          img.src = this.src;
-          this.loading = true;
-          img.onload = function() {
-            this._setSrc(this.src);
-            this.loading = false;
-            if (this.fade) {
-              this._placeholderEl.classList.add('fadein');
-            }
-          }.bind(this);
-        }
+    srcChanged: function(newSrc, oldSrc) {
+      if (newSrc !== oldSrc) {
+        this.loaded = false;
+      }
+    },
+
+    placeholderBackgroundUrlChanged: function() {
+      this.$.placeholder.style.backgroundImage =
+        this.placeholderBackgroundUrl;
+    },
+
+    transformChanged: function() {
+      var placeholderStyle = this.$.placeholder.style;
```

Comment:
In this particular case, turning the placeholder's `style` attribute into a computed binding will cause some duplication of effort. Changes to `sizing` and `position` have mirrored effects on `placeholder.style` AND `this.style`. However, the effects on the style attribute diverge for *one additional* property (`backgroundImage`). If we went the computed route, we would have two computed properties that mostly do the same thing (one for each of two `style` attributes).